packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -2534,3 +3049,664 @@ Future<TimeOfDay?> showTimePicker({
 void _announceToAccessibility(BuildContext context, String message) {
   SemanticsService.announce(message, Directionality.of(context));
 }
+
+// An abstract base class for the M2 and M3 defaults below, so that their return
+// types can be non-nullable.
+abstract class _TimePickerDefaults extends TimePickerThemeData {
```

Comment:
Interesting. In the other components we just went with a direct subclass of the component theme data. For the few cases where the default had to end in a non-null value, we would just use the '!' operator when pulling it from the defaults. Not as clean perhaps, but reduces the amount of bloat in having yet another full subclass definition in each component.

However, in this case it looks like you are also adding new fields that aren't part of the theme data. Is there some reason those new fields shouldn't be part of the public theme?

Reply:
I made the separate class with non-null fields so that I could use it everywhere as the type of the resolved default theme, then I wouldn't need to use the `!` everwhere that the default theme is used. It felt a lot cleaner to me, and the intermediate class is pretty much empty.

The sizes seemed like either a bunch of sizes that weren't part of the spec as being customizable, but were part of the token information, or which consolidated the sizing information that was encoded as a bunch of constants in the file that had to be different for M2 and M3, so it seemed better to just use the defaults for those, since they have to be different for M2 and M3 as well.

So, instead of:

```dart
const double _kDayPeriodInputWidth = xx;
const double _kDayPeriodInputHeight = xx;
const double _kDayPeriodInputWidthM2 = xx;
const double _kDayPeriodInputHeightM2 = xx;
```

I could just refer to the sizes in the defaults.

Reply:
That makes sense. Early on in the migration we got some push back on adding more classes to each widget as it was inflating the download size for the web. I believe each dart class gets transpiled into a JS class, so that could be significant for these boilerplate heavy classes. That said in this case it probably doesn't matter that much and ti is cleaner. Thx.